src/net/discovery/conf.rs

```diff
@@ -112,43 +53,26 @@ mod tests {
 
     #[test]
     fn parse_config() {
-        use config::{Config, File, FileFormat};
-
-        let to_conf: ToConf = Config::builder()
-            .add_source(File::from_str(EXAMPLE_CONFIG, FileFormat::Toml))
-            .build()
-            .unwrap()
-            .try_deserialize()
-            .expect("config should be valid");
-        let conf = Conf::from_file_conf(&to_conf).expect("file should contain valid values");
+        let conf = localhost_config([3000, 3001, 3002]);
 
         // H1
+        assert_eq!(conf.peers[Role::H1].origin, H1_URI.parse::<Uri>().unwrap());
```

Comment:
should we replace `Role::*` with just indexes? seems like there shouldn't be a concept of `Role` here at all, right?

Reply:
to summarize, we'll need 3 different kinds of mappings right?
* this mapping here, which is `endpoint -> http conf`
* mapping that report collector tells us, which is `secret_shares -> endpoint`
* mapping to assign roles, which is `endpoint -> role`

and only by combining all 3 can we accurately initialize a "ring"

Reply:
> should we replace Role::* with just indexes? seems like there shouldn't be a concept of Role here at all, right?

yes! Nothing in our code should remind us that helpers have hardcoded roles. 

> to summarize, we'll need 3 different kinds of mappings right?

that's right. Only one of those mappings is static which is the first one, two others are decided at runtime.

Reply:
👍